internal/venice-common/src/main/java/com/linkedin/venice/meta/ReadOnlySchemaRepository.java

```diff
@@ -68,4 +69,29 @@ public interface ReadOnlySchemaRepository extends VeniceResource {
   RmdSchemaEntry getReplicationMetadataSchema(String storeName, int valueSchemaId, int replicationMetadataVersionId);
 
   Collection<RmdSchemaEntry> getReplicationMetadataSchemas(String storeName);
+
+  default DerivedSchemaEntry getLatestDerivedSchema(String storeName) {
+    SchemaEntry valueSchemaEntry = getSupersetOrLatestValueSchema(storeName);
+    try {
+      return getLatestDerivedSchema(storeName, valueSchemaEntry.getId());
+    } catch (Exception e) {
+      /**
+       * Can't find the derived schema for the latest value schema, so it will fall back to find out the latest
+       * value schema, which has a valid derived schema.
+       */
+      Collection<DerivedSchemaEntry> derivedSchemaEntries = getDerivedSchemas(storeName);
+      DerivedSchemaEntry latestDerivedSchemaEntry = null;
+      for (DerivedSchemaEntry entry: derivedSchemaEntries) {
+        if (latestDerivedSchemaEntry == null || entry.getValueSchemaID() > latestDerivedSchemaEntry.getValueSchemaID()
+            || entry.getValueSchemaID() == latestDerivedSchemaEntry.getValueSchemaID()
+                && entry.getId() > latestDerivedSchemaEntry.getId()) {
```

Comment:
Minor nitpick on readability... I find it a bit hard to parse a sequence of mixed boolean operators... IIUC, `&&` has higher precedence than `||`, so the if condition `A || B || C && D` is equivalent to `A || B || (C && D)`. Although they are equivalent, I find the latter easier to parse visually. Or alternatively, the whole expression might be more concise and clear if using an extra function, e.g. `A || B || func(C, D)`, where `func(C, D) { return C && D; }`. What do you think?

Reply:
Sure, I will tweak the if condition.